pkg/ingestor/parser/cdx_vex/parser_cdx_vex.go

```diff
@@ -0,0 +1,207 @@
+	return &cdxVexParser{
+		identifierStrings: &common.IdentifierStrings{},
+	}
+}
+
+// Parse breaks out the document into the graph components
+func (c *cdxVexParser) Parse(ctx context.Context, doc *processor.Document) error {
+	c.doc = doc
+	err := json.Unmarshal(doc.Blob, &c.cdxBom)
+	if err != nil {
+		return fmt.Errorf("unable to unmarshal CDX VEX document: %w", err)
+	}
+
+	return nil
+}
+
+// GetIdentities gets the identity node from the document if they exist
+func (c *cdxVexParser) GetIdentities(ctx context.Context) []common.TrustInformation {
+	return nil
+}
+
+func (c *cdxVexParser) GetIdentifiers(ctx context.Context) (*common.IdentifierStrings, error) {
+	return nil, fmt.Errorf("not yet implemented")
+}
+
+func generateVexIngest(ctx context.Context) *assembler.VexIngest {
+	return nil
+}
+
+// Get package name and range versions to create package input spec for the affected packages.
+func getAffectedPackages(ctx context.Context, vulnInput *generated.VulnerabilityInputSpec, vexData generated.VexStatementInputSpec, affectsObj cdx.Affects) *[]assembler.VexIngest {
+	logger := logging.FromContext(ctx)
+	var pkgRef string
+	// TODO: retrieve purl from metadata if present - https://github.com/guacsec/guac/blob/main/pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go#L76
+	if affectsObj.Ref != "" {
+		pkgRef = affectsObj.Ref
+	} else {
+		logger.Warnf("[cdx vex] package reference not found")
+		return nil
+	}
+
+	// split ref using # as delimiter.
+	pkgRefInfo := strings.Split(pkgRef, "#")
+	if len(pkgRefInfo) != 2 {
+		logger.Warnf("[cdx vex] malformed package reference: %q", affectsObj.Ref)
+		return nil
+	}
+	pkgURL := pkgRefInfo[1]
+
+	// multiple package versions do not exist, resolve to using ref directly.
+	if affectsObj.Range == nil {
+		pkg, err := helpers.PurlToPkg(pkgURL)
+		if err != nil {
+			logger.Warnf("[cdx vex] unable to create package input spec: %v", err)
+			return nil
+		}
+
+		return &[]assembler.VexIngest{{VexData: &vexData, Vulnerability: vulnInput, Pkg: pkg}}
+	}
+
+	// split pkgURL using @ as delimiter.
+	pkgURLInfo := strings.Split(pkgURL, "@")
+	if len(pkgURLInfo) != 2 {
+		logger.Warnf("[cdx vex] malformed package url info: %q", pkgURL)
+		return nil
+	}
+
+	pkgName := pkgURLInfo[0]
+	var viList []assembler.VexIngest
+	for _, affect := range *affectsObj.Range {
+		// TODO: Handle package range versions (see - https://github.com/CycloneDX/bom-examples/blob/master/VEX/CISA-Use-Cases/Case-8/vex.json#L42)
+		if affect.Version == "" {
+			continue
+		}
+		vi := &assembler.VexIngest{
+			VexData:       &vexData,
+			Vulnerability: vulnInput,
+		}
+
+		pkg, err := helpers.PurlToPkg(fmt.Sprintf("%s@%s", pkgName, affect.Version))
+		if err != nil {
+			logger.Warnf("[cdx vex] unable to create package input spec from purl: %v", err)
+			return nil
+		}
+		vi.Pkg = pkg
+		viList = append(viList, *vi)
+	}
+
+	return &viList
+}
+
+func (c *cdxVexParser) GetPredicates(ctx context.Context) *assembler.IngestPredicates {
```

Comment:
https://raw.githubusercontent.com/CycloneDX/bom-examples/master/VEX/vex.json also contains scores, this can now be captured via the new [vuln metadata](https://github.com/guacsec/guac/blob/main/pkg/assembler/assembler.go#L103-L107) nodes